File: java/src/main/java/AppTestingIOS.java
import org.openqa.selenium.remote.DesiredCapabilities;
import org.openqa.selenium.TakesScreenshot;
import org.openqa.selenium.OutputType;

import io.appium.java_client.MobileElement;
import io.appium.java_client.android.AndroidDriver;

import java.net.URL;
import java.util.Set;

public class AppTestingIOS {
  public static final String USERNAME = System.getenv("BROWSERSTACK_USER");
  public static final String AUTOMATE_KEY = System.getenv("BROWSERSTACK_ACCESS_KEY");
  public static final String URL = "http://" + USERNAME + ":" + AUTOMATE_KEY + "@hub-cloud.browserstack.com/wd/hub";

  public static void main(String[] args) throws Exception {
    
    DesiredCapabilities caps = new DesiredCapabilities();
    caps.setCapability("browserstack.debug", "true");
    caps.setCapability("build", "App Testing iOS");
    caps.setCapability("name", "sample java");
    caps.setCapability("platformName", "ios");
    caps.setCapability("device", "iPhone 6S");
    caps.setCapability("realMobile", true);
    caps.setCapability("app", System.getenv("IOS_APP_URL"));

    AndroidDriver driver = new AndroidDriver(new URL(URL), caps);

    ((TakesScreenshot)driver).getScreenshotAs(OutputType.BASE64);

    Set<String> contexts = driver.getContextHandles();

    MobileElement num1 = (MobileElement) driver.findElementById("IntegerA");
    MobileElement num2 = (MobileElement) driver.findElementById("IntegerB");
    num1.sendKeys("12");
    num2.sendKeys("32");

    MobileElement add_btn = (MobileElement) driver.findElementById("ComputeSumButton");
    driver.tap(1, add_btn, 1000);

    ((TakesScreenshot)driver).getScreenshotAs(OutputType.BASE64);
    MobileElement sampleLabel = (MobileElement) driver.findElementById("Answer");
    System.out.println(sampleLabel.getText());

    driver.quit();
  }
}

File: python/app-testing-ios.py
import os

from appium import webdriver
from appium.webdriver.common.touch_action import TouchAction

desired_cap = { 
  'platformName': 'ios',
  'device': 'iPhone 6S',
  'realMobile': True,
  'app': os.environ['IOS_APP_URL'],
  'build': 'App Testing iOS',
  'name': 'sample python',
  'browserstack.debug': True
}

driver = webdriver.Remote(
  command_executor='http://'+os.environ['BROWSERSTACK_USER']+':'+os.environ['BROWSERSTACK_ACCESS_KEY']+'@hub.browserstack.com:80/wd/hub',
  desired_capabilities=desired_cap)


driver.get_screenshot_as_file("test.png")
context = driver.contexts

num1 = driver.find_element_by_id("IntegerB")
num2 = driver.find_element_by_id("IntegerB")
num1.send_keys("12")
num2.send_keys("32")
add_btn = driver.find_element_by_id("ComputeSumButton")

touch = TouchAction(driver)
touch.tap(add_btn).perform()

driver.get_screenshot_as_file("test.png")
sample_label = driver.find_element_by_id("Answer")
print(sample_label.text)

driver.quit()
